mypy/checker.py

```diff
@@ -2444,7 +2444,7 @@ def type_is_iterable(self, type: Type) -> bool:
             type = type.fallback
         return (is_subtype(type, self.named_generic_type('typing.Iterable',
                                                          [AnyType(TypeOfAny.special_form)])) and
-                isinstance(type, Instance))
+                isinstance(type, (Instance, TupleType)))
```

Comment:
But why this is needed at all? What happens if you remove this `and ...`? This deserves at least a TODO (or better actually fixing it).

Reply:
It looks like this `Instance` check was needed at the only other call site, in `check_multi_assignment_from_iterable` (right below). I moved this check out of the function call, since it seems to be a separate concern.